src/structures/AnonymousGuild.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const BaseGuild = require('./BaseGuild');
+const { VerificationLevels, NSFWLevels } = require('../util/Constants');
+
+/**
+ * Bundles common attributes between guild classes
+ * @abstract
+ */
+class AnonymousGuild extends BaseGuild {
+  constructor(client, data) {
+    super(client, data);
+    this._patch(data);
+  }
+
+  _patch(data) {
+    this.features = data.features;
+    /**
+     * The hash of the guild invite splash image
+     * @type {?string}
+     */
+    this.splash = data.splash;
+
+    /**
+     * The hash of the guild banner
+     * @type {?string}
+     */
+    this.banner = data.banner;
+
+    /**
+     * The description of the guild, if any
+     * @type {?string}
+     */
+    this.description = data.description;
+
+    /**
+     * The verification level of the guild
+     * @type {VerificationLevel}
+     */
+    this.verificationLevel = VerificationLevels[data.verification_level];
+
+    /**
+     * The vanity invite code of the guild, if any
+     * @type {?string}
+     */
+    this.vanityURLCode = data.vanity_url_code;
+
+    if ('nsfw_level' in data) {
+      /**
+       * The NSFW level of this guild
+       * @type {NSFWLevel}
```

Comment:
Possibly undefined

Reply:
ported this from Guild as-is, assigns when the data is there (which will be the case on initial payloads) doesn't touch it if not present in data (on partial updates that don't change this field)